extensions/qcom/GLSL_QCOM_tile_shading.txt

```diff
@@ -0,0 +1,326 @@
+
+    tile_attachmentQCOM                          X          X            X        -      fragment/compute
+                                                                                         uniform (Vulkan only)
+
+    shading_rate_xQCOM =                         X                                       compute in
+    shading_rate_yQCOM =                                                                 (Vulkan only)
+    shading_rate_zQCOM =
+
+
+    Modify the Fragment Shader Inputs subsection under 4.4.1 Input Layout Qualifiers (p. 76)
+
+    Fragment shaders allow the following layout qualifier of in only (not with variable declarations):
+
+    layout-qualifier-id:
+        early_fragment_tests
+        non_coherent_attachment_readQCOM
+
+    The early_fragment_tests qualifier is used to request that fragment tests be
+    performed before fragment shader execution, as described in section 15.2.4
+    "Early Fragment Tests" of the OpenGL Specification.
+
+
+    (add to the end of the Fragment Shader Inputs subsection).
+
+    The non_coherent_attachment_readQCOM qualifier requests that tile image attachment
+    reads ignore rasterization order for color/depth/stencil/input attachments.
+
+    For example, 
+     layout( non_coherent_attachment_readQCOM ) in;
+
+    Only one fragment shader (compilation unit) need declare this, though more
+    than one can. If at least one declares the non_coherent_attachment_readQCOM
+    qualifier, then the non-coherent attachment read is enabled.
+
+    See the Vulkan API documentation for details on rasterization order.
+
+
+    Modify the Compute Shader Inputs subsection under 4.4.1 Input Layout Qualifiers (p. 78)
+
+    There are no layout location qualifiers for compute shader inputs.
+
+    Layout qualifier identifiers for compute shader inputs are the workgroup size qualifiers:
+
+    layout-qualifier-id :
+        local_size_x = layout-qualifier-value
+        local_size_y = layout-qualifier-value
+        local_size_z = layout-qualifier-value
+        shading_rate_xQCOM = layout-qualifier-value
+        shading_rate_yQCOM = layout-qualifier-value
+        shading_rate_zQCOM = layout-qualifier-value
+
+
+    (insert the following three paragraphs before the sentence starting with "If multiple
+    compute shaders attached to a single program object" in the Compute Shader Inputs subsection).
+
+    The shading_rate_xQCOM, shading_rate_yQCOM, and shading_rate_zQCOM qualifiers are used
+    to declare a desired shading rate within a compute shader in the first, second,
+    and third dimension, respectively. If a shader does not specify a rate for one
+    of the dimensions, that dimension will have a rate of 1.
+    
+    The shading_rate_xQCOM and shading_rate_yQCOM qualifiers are required to be a power of 2.
+    If any of the shading rate values is greater than the maximum value supported by the 
+    implementation for that dimension, a run-time validation failure will result.
```

Comment:
You say what happens if the rate is over the maximum but only that a power of 2 is required. Is it a compile error to specify something else?

Reply:
```suggestion
    The shading_rate_xQCOM and shading_rate_yQCOM qualifiers are required to be a power of 2, otherwise a compile-time error results.
    If any of the shading rate values is greater than the maximum value supported by the 
    implementation for that dimension, a run-time validation failure will result.
```

Reply:
Done